sale_order_net_price_control/models/sale_order_line.py

```diff
@@ -0,0 +1,53 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, _, fields
+from odoo.exceptions import ValidationError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    offer_price = fields.Boolean()
+    admits_offer_prices = fields.Boolean(related='product_template_id.admits_offer_prices', store=True)
+
+    # FASE 1 desarrollo
+    def _compute_amount(self):
+        super()._compute_amount()
+        # FASE 3 DEL DESARROLLO
+        for record in self:
+            if not self.env.user.has_group('sale_order_net_price_control.group_sale_at_any_price'):
+                if record.product_id and record.product_uom_qty > 0:
+
+                    net_price_sale = record.price_subtotal / record.product_uom_qty
+
+                    purchase_invoice = self.env['account.move'].search([
+                        ('invoice_line_ids.product_id', '=', record.product_id.id),
+                        ('invoice_line_ids.quantity', '>', 0),
+                        ('move_type', '=', 'in_invoice'), # factura de compra
+                        ('state', '=', 'posted'), # confirmado
+                    ], limit=1, order='name desc')
+
+                    if purchase_invoice:
+                        invoice_line = purchase_invoice.invoice_line_ids.filtered(
+                        lambda x: x.product_id.id == self.product_id.id
+                        )
+                        net_price_purchase = invoice_line.price_subtotal / invoice_line.quantity
+
+                        if net_price_sale <= net_price_purchase:
+                            raise ValidationError(_(
+                                "Net price of '%s', '%.2f' must be higher that the last purchase invoice price '%s'"
+                            )%(record.product_template_id.name,net_price_sale,invoice_line.price_unit) )
+                        # FASE 4 DEL DESARROLLO - parte 1
+                        if not self.offer_price:
+                            if net_price_sale <= self._default_lowest_pricelist_id():
+                                raise ValidationError(_(
+                                    "Net price of '%s', '%.2f' must be higher that the MAYM pricelist '%s'"
+                                )%(record.product_template_id.name,net_price_sale,self._default_lowest_pricelist_id()) )
+
+    # FASE 2 desarrollo
+    def _default_lowest_pricelist_id(self):
+        company = self.env["res.company"].browse(
+            self._context.get("company_id", self.env.user.company_id.id)
+        )
```

Comment:
Good point taking in account company context, anyway this is a little bit inefficent. When `company_id` context value doesn't exist we don't need to use `browse()`, because `self.env.user.company_id` is the final company itself. And method name is incorrectly, because this returns a price, not a pricelist, please rename it

Not tested:
```suggestion
    def _get_price_from_lowest_pricelist_id(self):
        company = self._context.get("company_id", False) and self.env["res.company"].browse(
            self._context["company_id"]
        ) or self.env.user.company_id
```